ambry-tools/src/main/java/com.github.ambry/store/StoreCopier.java

```diff
@@ -0,0 +1,271 @@
+    private final InputStream stream;
+
+    /**
+     * @param messageInfo the {@link MessageInfo} for this message.
+     * @param stream the {@link InputStream} that represents the data of this message.
+     */
+    public Message(MessageInfo messageInfo, InputStream stream) {
+      this.messageInfo = messageInfo;
+      this.stream = stream;
+    }
+
+    /**
+     * @return the {@link MessageInfo} for this message.
+     */
+    public MessageInfo getMessageInfo() {
+      return messageInfo;
+    }
+
+    /**
+     * @return the {@link InputStream} that represents the data of this message.
+     */
+    public InputStream getStream() {
+      return stream;
+    }
+  }
+
+  /**
+   * An interface for a transformation function. Transformations can modify any data in the message (including keys).
+   */
+  public interface Transformer {
+    Message transform(Message message);
+  }
+
+  /**
+   * Config class for the {@link StoreCopier}.
+   */
+  private static class CopierConfig {
+
+    /**
+     * The path to the hardware layout file.
+     */
+    @Config("hardware.layout.file.path")
+    final String hardwareLayoutFilePath;
+
+    /**
+     * The path to the partition layout file.
+     */
+    @Config("partition.layout.file.path")
+    final String partitionLayoutFilePath;
+
+    /**
+     * The path of the directory where the source store files are.
+     */
+    @Config("src.store.dir")
+    final String srcStoreDirPath;
+
+    /**
+     * The path of the directory where the target store files should be.
+     */
+    @Config("tgt.store.dir")
+    final String tgtStoreDirPath;
+
+    /**
+     * The total capacity of the store.
+     */
+    @Config("store.capacity")
+    final long storeCapacity;
+
+    /**
+     * The size of each fetch from the source store.
+     */
+    @Config("fetch.size.in.bytes")
+    @Default("4 * 1024 * 1024")
+    final long fetchSizeInBytes;
+
+    CopierConfig(VerifiableProperties verifiableProperties) {
+      hardwareLayoutFilePath = verifiableProperties.getString("hardware.layout.file.path");
+      partitionLayoutFilePath = verifiableProperties.getString("partition.layout.file.path");
+      srcStoreDirPath = verifiableProperties.getString("src.store.dir");
+      tgtStoreDirPath = verifiableProperties.getString("tgt.store.dir");
+      storeCapacity = verifiableProperties.getLong("store.capacity");
+      fetchSizeInBytes = verifiableProperties.getLongInRange("fetch.size.in.bytes", 4 * 1024 * 1024, 1, Long.MAX_VALUE);
+    }
+  }
+
+  private static final Logger logger = LoggerFactory.getLogger(StoreCopier.class);
+
+  private final String storeId;
+  private final Store src;
+  private final Store tgt;
+  private final long fetchSizeInBytes;
+  private final List<Transformer> transformers;
+  private final ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+  private boolean isOpen = true;
+
+  public static void main(String[] args) throws Exception {
```

Comment:
Yes, the `StoreCopier` can be used on its own to reformat single stores. The `DiskReformatter` is a wrapper that works on all stores in a disk.